src/rt/critical_.d

```diff
@@ -25,7 +25,8 @@ extern (C) void _d_critical_init()
 
 extern (C) void _d_critical_term()
 {
-    for (auto p = head; p; p = p.next)
+    auto h = cast()head;
+    for (auto p = h; p; p = p.next)
```

Comment:
Should this struct really be `shared` ?

Reply:
I think so, see `ensureMutex` just below.